File: BackgroundVideoDemo/BackgroundVideoDemo/ViewController.m
//
//  ViewController.m
//  BackgroundVideoDemo
//
//  Created by Adam Albarghouthi on 2016-06-26.
//  Copyright © 2016 backgroundVideo. All rights reserved.
//

#import "ViewController.h"
#import "BackgroundVideoObjC.h"

@interface ViewController ()

@property (strong, nonatomic) BackgroundVideoObjC *backgroundVideo;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.backgroundVideo = [[BackgroundVideoObjC alloc] initOnViewController:self withVideoURL:[NSURL URLWithString:@"test.mp4"]];
    [self.backgroundVideo setUpBackground];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Demo-Swift/Demo/ViewController.swift
//
//  ViewController.swift
//  Demo
//
//  Created by Amr Guzlan on 2016-06-26.
//  Fork modified by Doron Katz on 2017-03-03
//  Copyright © 2016 Amro Gazlan. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    var backgroundPlayer : BackgroundVideo? // Declare an instance of BackgroundVideo called backgroundPlayer

    override func viewDidLoad() {
        super.viewDidLoad()
        // Initializing your instance
        do{
            try backgroundPlayer = BackgroundVideo(on: self.view, withVideoURL: "test.mp4") // Passing self and video name with extension
            try backgroundPlayer?.setUpBackground()
        }catch BackgroundVideoErrors.invalidVideo(message: "Video is invalid"){
            print("Video is invalid")
        }catch BackgroundVideoErrors.missingVideo(message: "View is missing"){
            print("Video is missing")
        }catch{
            print("Unknown error")
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

